draft-irtf-nwcrg-coding-and-congestion-01.txt

```diff
@@ -1,504 +1,528 @@
+
+Internet-Draft           Coding and congestion              January 2020
+
+Copyright Notice
+
+   Copyright (c) 2020 IETF Trust and the persons identified as the
+   document authors.  All rights reserved.
+
+   This document is subject to BCP 78 and the IETF Trust's Legal
+   Provisions Relating to IETF Documents (http://trustee.ietf.org/
+   license-info) in effect on the date of publication of this document.
+   Please review these documents carefully, as they describe your rights
+   and restrictions with respect to this document.  Code Components
+   extracted from this document must include Simplified BSD License text
+   as described in Section 4.e of the Trust Legal Provisions and are
+   provided without warranty as described in the Simplified BSD License.
+
+Table of Contents
+
+   1.  Introduction . . . . . . . . . . . . . . . . . . . . . . . . .  2
+   2.  Separate channels, separate entities . . . . . . . . . . . . .  3
+   3.  Scope  . . . . . . . . . . . . . . . . . . . . . . . . . . . .  4
+     3.1.  Type of application  . . . . . . . . . . . . . . . . . . .  4
+     3.2.  End-to-end . . . . . . . . . . . . . . . . . . . . . . . .  4
+   4.  Base solution description  . . . . . . . . . . . . . . . . . .  4
+   5.  Server-side coding solutions . . . . . . . . . . . . . . . . .  5
+     5.1.  Coded packets without considering CWND progression . . . .  5
+     5.2.  Coded packets driven by CWND progression . . . . . . . . .  5
+   6.  Server-side reaction to repaired packet signals  . . . . . . .  6
+     6.1.  The server congestion control considers repaired packet
+           signals as congestion-implied packet losses  . . . . . . .  6
+     6.2.  The server adapts its window reduction to repaired packet
+           signals  . . . . . . . . . . . . . . . . . . . . . . . . .  6
+     6.3.  The server ignores repaired packet signals . . . . . . . .  7
+   7.  Summary  . . . . . . . . . . . . . . . . . . . . . . . . . . .  7
+   8.  Acknowledgements . . . . . . . . . . . . . . . . . . . . . . .  8
+   9.  IANA Considerations  . . . . . . . . . . . . . . . . . . . . .  8
+   10. Security Considerations  . . . . . . . . . . . . . . . . . . .  8
+   11. References . . . . . . . . . . . . . . . . . . . . . . . . . .  8
+   Authors' Addresses . . . . . . . . . . . . . . . . . . . . . . . .  9
+
+1.  Introduction
+
+   This document represents the collaborative work and consensus of the
+   Coding for Efficient Network Communications Research Group (NWCRG);
+   it is not an IETF product and is not a standard.
+
+
+
+
+
+
+
+
+
+
+
+
+Kuhn, Lochin, Michel & WeExpires July 11, 2020                  [Page 2]
+
+Internet-Draft           Coding and congestion              January 2020
+
+
+   There are cases where deploying coding improves the quality of the
+   transmission.  As an example, the server may hardly detect tail
+   losses that impact may impact the application layer [MICHAEL: I don't

```

Comment:
You mean an application on top of a congestion controller, I guess?
Then: if it assumes reliable transfer, it won't see the loss, it will just see delay (or a lower throughput on average). If it doesn't assume reliable transfer, it will take just as long to notice it, but coding can mask the tail loss. The latter is a good thing, but the statement about the server detecting tail loss late is confusing either way.

If you don't mean an application on top of a congestion controller, I don't understand this at all. A congestion control mechanism will not detect any loss earlier due to coding.

Reply:
I just mean that tail losses often take longer to be detected and therefore the transfer will also take longer to be completed (sorry I should have stated that in my comment). Using FEC allows to improve the download time by allowing to repairing these losses directly so that you can deliver more rapidly the data to the underlying application.

Even if it could (e.g. in QUIC), the idea here is not to say that the losses will be detected earlier but that they will be repaired earlier (before their detection), allowing to deliver the end of the data faster to the application.
 So I propose we rephrase the confusing statement that way ?

Reply:
I get it, tail losses take longer than other losses to be detected. So, particularly for these, coding helps (and I agree). Yes, let's rephrase ... or re-write, in the new structure.

Reply:
This is it. Tail losses take longer to be detected and could seriously harm the QoE in long delay networks.